drivers/volume/portworx/portworx.go

```diff
@@ -625,6 +627,81 @@ func (d *portworx) WaitForUpgrade(n node.Node, image, tag string) error {
 	return nil
 }
 
+func (d *portworx) GetReplicationFactor(vol *torpedovolume.Volume) (int64, error) {
+	name := d.schedOps.GetVolumeName(vol)
+	t := func() (interface{}, bool, error) {
+		vols, err := d.volDriver.Inspect([]string{name})
+		if err != nil && err == volume.ErrEnoEnt {
+			return 0, false, volume.ErrEnoEnt
+		} else if err != nil {
+			return 0, false, err
+		}
+		if len(vols) == 1 {
+			return vols[0].Spec.HaLevel, false, nil
+		}
+		return 0, false, fmt.Errorf("Extra volumes with the same volume name/ID seen") //Shouldn't reach this line
+	}
+
+	tmp, err := task.DoRetryWithTimeout(t, defaultRetryInterval, defaultRetryInterval)
+	if err != nil {
+		return 0, &ErrFailedToGetReplicationFactor{
+			ID:    name,
+			Cause: err.Error(),
+		}
+	}
+	repFactor, ok := tmp.(int64)
+	if !ok {
+		return 0, &ErrFailedToGetReplicationFactor{
+			ID:    name,
+			Cause: fmt.Sprintf("Replication factor is not of type int64"),
+		}
+	}
+
+	return repFactor, nil
+}
+
+func (d *portworx) SetReplicationFactor(vol *torpedovolume.Volume, replFactor int64) error {
+	name := d.schedOps.GetVolumeName(vol)
+	t := func() (interface{}, bool, error) {
+		vols, err := d.volDriver.Inspect([]string{name})
+		if err != nil && err == volume.ErrEnoEnt {
+			return nil, false, volume.ErrEnoEnt
+		} else if err != nil {
+			return nil, false, err
+		}
+		spec := &api.VolumeSpec{
```

Comment:
we should get the existing spec from vols[0] and then only change the HaLevel, instead reinitializing the object, right?

also move this under -  if len(vols)== 1